pkg/preparer/setup.go

```diff
@@ -545,28 +553,51 @@ func New(preparerConfig *PreparerConfig, logger logging.Logger) (*Preparer, erro
 
 	podFactory := pods.NewFactory(preparerConfig.PodRoot, preparerConfig.NodeName, fetcher, preparerConfig.RequireFile, readOnlyPolicy)
 	podFactory.SetOSVersionDetector(osVersionDetector)
+
+	// TODO: we might want to customize our docker client
+	dockerClient, err := dockerclient.NewEnvClient()
+	if err != nil {
+		return nil, util.Errorf("could not create docker client: %s", err)
+	}
+
+	containerRegistryAuthStr := ""
+	if preparerConfig.ContainerRegistryJsonKeyFile != "" {
+		data, err := ioutil.ReadFile(preparerConfig.ContainerRegistryJsonKeyFile)
+		if err != nil {
+			return nil, util.Errorf("could not read container json key file %s: %s", preparerConfig.ContainerRegistryJsonKeyFile, err)
+		}
+		authConfig := dockertypes.AuthConfig{
+			Username: "_json_key",
+			Password: string(data),
+		}
+		encodedJSON, err := json.Marshal(authConfig)
```

Comment:
check `err`. Also this code looks the same as what's in `p2-launch`, can it be centralized into a library somewhere?